### PR TITLE
Add simple test for text, vision, tabular and collab for model summary

### PR DESCRIPTION
**Pull request by noklam**
Previous PR for vision summary: https://github.com/fastai/fastai/pull/1198
Dev Project Index: https://forums.fast.ai/t/dev-projects-index/29849
Discussion thread on Forum: https://forums.fast.ai/t/model-summary-implementation/31193

**Plan for this PR**  
add:
* units tests for model_summary
* support for more fastai learner type
* support for pure PyTorch nn.Module
* model_summary takes both pure nn.Module and fastai Learner

**Plan for future PR**  

add:
* more information like trainable parameters
* graph visualization for model structure
* Backbone, head structure shown in summary

- [x] Add simple unit test for text, vision tabular and collab Learner
- [x] Add simple unit test for nn.Module
- [x] model_summary work for nn.Module and subclass of nn.Module consists multiple nn.Module
- [x] model summary work for all fastai Learner subclass
- [x] Combine them to have 1 model_summary work for all learner

 Note: Still not working for Text model, but other learner are fine.

### CODE REVIEW

**Comment by noklam** [comment]
Progress: work for standard pytorch module and most fastai Learner except Text now.

**Comment by noklam** [comment]
Haven't figure out how to make text model work.

File: fastai/callbacks/hooks.py [hunk 1]
```diff
@@ -99,19 +99,29 @@ def num_features_model(m:nn.Module)->int:
...
+    if hasattr(m, "bias") and hasattr(m.bias, "size"): params += m.bias.numel()
     return params
 
 def hook_params(modules:Collection[nn.Module]) -> Hooks:
     return Hooks(modules, lambda m, i, o: total_params(m))
 
 def params_size(m: nn.Module, size: tuple = (64, 64)) -> Tuple[Sizes, Tensor, Hooks]:
     "Pass a dummy input through the model to get the various sizes. Returns (res,x,hooks) if `full`"
+    if isinstance(m, Learner):
+        dl = m.data.train_dl
```
Comment (sgugger) [inline fastai/callbacks/hooks.py]: You have method one_batch in DataBunch that will do this for you now.

Comment (noklam) [reply fastai/callbacks/hooks.py]: Thanks!

**Comment by noklam** [comment]
The support for text model will be added in future PR, I think this one is ok to be merged

**Pull request merged**